[teamotitan] iP

teamotitan opened the pull request:
# TimoPro
> #### "Life is a paradise of endless possibilities!" - [Mehmet Murat ildan](https://quotefancy.com/quote/2348216/Mehmet-Murat-ildan-Life-is-a-paradise-of-endless-possibilities)

TimoPro frees your mind of having to remember to many things that you need to do every single day! It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1) download it from [here](https://github.com/teamotitan/ip/releases/tag/A-Jar)
2) double-click it.
3) add your tasks.
4) let it manage your tasks for you! 👍 

And it is **FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method

```java
public class Main {
    public static void main(String[] args) {
         Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

teamotitan committed:
Level 0

teamotitan committed:
level 1

teamotitan committed:
Level-2

teamotitan committed:
level 3

teamotitan committed:
Level 4

teamotitan committed:
A-TextUiTesting

teamotitan committed:
Level 5

teamotitan committed:
level 6

teamotitan committed:
Level 7

teamotitan committed:
Merge branch 'branch-Level-7'

teamotitan committed:
Level 8

teamotitan committed:
Merge branch 'branch-level-8'

teamotitan committed:
Level A-MoreOOP

teamotitan committed:
Level A-Packages

teamotitan committed:
Merge branch 'add-gradle-support'

teamotitan committed:
A-Gradle

teamotitan committed:
A-JUnit

teamotitan committed:
A-Jar

teamotitan committed:
A-JavaDoc

teamotitan committed:
A-javaDoc

teamotitan committed:
A-CodingStandard

teamotitan committed:
level-9

teamotitan committed:
Merge branch 'branch-A-JavaDoc'

teamotitan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Timo/Timo.java

teamotitan committed:
Merge branch 'branch-Level-9'

rayray39 commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+import java.time.format.DateTimeFormatter;
+
+class Task {
+    private String taskype = "T";
+    private boolean mark;
```
Is there a typo in ```String taskype```? Perhaps ```boolean mark``` could sound more like a boolean, for example adding a prefix such as is, can, has etc.

teamotitan replied:
Okay thanks for pointing out my mistakes. I will go and resolve them!

rayray39 commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+    private String taskype = "T";
+    private boolean mark;
+    private String val;
+    public Task(boolean mark, String val) {
```
Perhaps something more descriptive could be used in place of ```val```?

teamotitan replied:
Ok that's a good idea, thanks!

rayray39 commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+                        arr.add(new Todo(false, a[1]));
+                    }
+                } else if (tmp.startsWith("[D]")) {
+                    //remove the [D][?] from the line
```
I like how comments are added to facilitate understanding how the code creates a deadline and event, perhaps similar comments could be added into todo as well?

teamotitan replied:
Ok I will do that, thanks!

rayray39 commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+
+            default:
+                String s = String.valueOf(command.charAt(command.length() - 1));
+                if (command.startsWith("mark")) {
```
Any reason why mark, unmark, delete, todo, deadline, event and find are not cases of the switch block?

teamotitan replied:
You are right, I will go edit it!

rayray39 commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+        //print list initially
+        ui.printList(this.tasks);
+
+        boolean isExit = false;
```
Good use of a boolean variable that sounds like booleans

teamotitan replied:
Thank you!

rayray39 reviewed (commented):
Overall the code was easy to understand and has a logical flow. Consider splitting the classes into different files if that makes sense.

yanqiyqh commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+
+}
+
+//Timo.Storage: beginning I read from the file, end I update the file
```
Maybe this can be worded better

yanqiyqh commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+        return (this.mark ? "X" : " ");
+    }
+
+    @Override
```
Nice that you included @Override tags

yanqiyqh commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+import java.io.File;
+import java.time.format.DateTimeFormatter;
+
+class Task {
```
Consider separating the classes into different files? e.g. Task.java, Todo.java etc

yanqiyqh commented on src/main/java/Timo/Timo.java:
```diff
@@ -0,0 +1,584 @@
+class Event extends Task {
+    private String from;
+    private String to;
+    private String tasktype = "E";
```
Since tasktype is tied to the Class and does not change, it should be a static variable: `private static String TASKTYPE = "E";`

yanqiyqh reviewed (commented):
Overall, your code largely follows the coding standards. Nice!

teamotitan committed:
A-CheckStyle

teamotitan committed:
Merge tag 'A-CheckStyle'

teamotitan committed:
Level-10

teamotitan committed:
Merge tag 'Level-10'

teamotitan committed:
Update README.md

teamotitan committed:
Update README.md

teamotitan committed:
Currently when loading the lines from the list text file, the else
statement assumes that the task is an event and does not check
whether it is in fact an event.

Similarly, when loading the lines from the text, I only check that
it is marked, and the other case assumed that the task is unmarked

I did not account for the case where task might not be unmarked, but
rather other characters are placed inside, or it might not be an
event, but rather some other characters, which can happen if users
tamper with the list text file manually

Two assert have been added, to check if it is a task is an event,
and to check if the task is unmarked, instead of assuming

teamotitan committed:
Currently, there are code that are not in use since we decided
to switch to GUI, such as greeting the user function, and run
function in Timo

Moreover, the case in Parser are all over the place, such as bye
case being at the top, todo, event and deadline cases being below
mark and unmark, making the cases quite confusing to look at

Lastly, when I load the data from the text file to the array, the
load function is too long

I have removed code that are no longer needed, as well as adjusted
the ordering of the cases in the parser, as well as changed
variable name of DateTimeFormatter from "a" to inputFormat. In
addition, I have also added a method to addTaskToArray to make the
load function more clean

teamotitan committed:
no message

teamotitan committed:
Merge pull request #1 from teamotitan/branch-A-Assertions

Added Assertions

teamotitan committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Timo.java

teamotitan committed:
Merge pull request #2 from teamotitan/branch-A-CodeQuality

Branch a code quality

teamotitan committed:
no message

teamotitan committed:
BCD-Extension

teamotitan committed:
BCD-Extension

teamotitan committed:
Merge branch 'branch-Undo'

teamotitan committed:
Merge branch 'master' of https://github.com/teamotitan/ip

teamotitan committed:
no message

teamotitan committed:
added in user guide

teamotitan committed:
no message

teamotitan committed:
Update README.md

teamotitan committed:
Went for smoke testing, as well as testing by other groupmates

Found multiple errors which included,
if I todo bye, the program will exit
if my todo, deadline and event description are empty it will still
be accepted
if my index is 0 for mark, unmark and delete, there will be error
but they wont show the error in the UI

Fixed these bugs

teamotitan committed:
Merge branch 'master' of https://github.com/teamotitan/ip

teamotitan committed:
Made minor fixes such as making boolean sound more boolean, as well as
fixing the bug where the data is not saved

teamotitan committed:
Removed the hashes

teamotitan committed:
Made additional changed to README.md

teamotitan committed:
Edit README.md

In README.md, example output: it is written like //example output

It does not bold the words and readers might miss the words

Let's change the example output to be bold

teamotitan committed:
MainWindow.java: Edited Ui output

teamotitan committed:
MainWindow.java: Edited Ui output

teamotitan committed:
MainWindow.java: Changed program exit delay time

teamotitan committed:
MainWindow.java: Edited Ui output

teamotitan committed:
Edit Ui.java

teamotitan committed:
JUnit tests: created more tests

JUnit tests were commented out

there were no tests to test the program

Let's,
* create JUnit tests to test the program

teamotitan committed:
Timo package: change to match style

The naming of the package did not match naming conventions

Let's,
*change the naming of the package to match the conventions

teamotitan committed:
Jar file: make it store data

Previously, when the Jar file was created, after exiting the
application the list was not stored

Let's
* edit the Storage.java file so that the file will be created

teamotitan committed:
Fixed too long methods

The methods that were written were too long

Let's
* shorten the methods so that each method is less than 30LoC each
